v2/cli/init_command.go

```diff
@@ -0,0 +1,255 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/brigadecore/brigade/v2/internal/file"
+	"github.com/urfave/cli/v2"
+)
+
+type initFields struct {
+	ProjectID   string
+	Language    string
+	GitCloneURL string
+	Script      string
+}
+
+var initCommand = &cli.Command{
+	Name:  "init",
+	Usage: "Initialize a new Brigade project",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:    flagID,
+			Aliases: []string{"i"},
+			Usage: "Name of the Brigade project to initialize " +
+				"(required)",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:    flagLanguage,
+			Aliases: []string{"l"},
+			Usage: "Specify the scripting language to use for handling events" +
+				" -- JavaScript (js) or TypeScript (ts)",
+			Value: "ts",
+		},
+		&cli.StringFlag{
+			Name:    flagGit,
+			Aliases: []string{"g"},
+			Usage: "Optionally specify a git repository where the" +
+				" event handling script can be found",
+		},
+	},
+	Action: initialize,
+}
+
+// Handle init CLI flags
+// nolint: gocyclo
+func initialize(c *cli.Context) error {
+
+	// Initialize fields to fill in the template yaml files
+	fields := initFields{
+		ProjectID:   c.String(flagID),
+		Language:    strings.ToLower(c.String(flagLanguage)),
+		GitCloneURL: c.String(flagGit),
+	}
+
+	err := idMatch(fields.ProjectID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = canonicalizeLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	err = checkCreateDirectory(".brigade")
+	if err != nil {
+		return err
+	}
+
+	// Generate js/ts buffer
+	var scriptBytes []byte
+	scriptBytes, err = execTemplate(scriptTemplate, fields)
+	if err != nil {
+		return err
+	}
+	fields.Script = string(scriptBytes)
+
+	brigadeScriptFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+	// If git repo is provided, create a separate js/ts file for script
+	var overwrite bool
+	scriptPath := path.Join(".brigade", brigadeScriptFileName)
```

Comment:
As a general rule, try to give most variables the narrowest scope possible. In this case, these can be moved into the condition block that follows.

Reply:
`overwrite` is used again on line 116 in another conditional, so would it make more sense to leave the declaration of `overwrite` alone?

Reply:
Sure. But move the other one.